File: internal/cmd/manifest.go
package cmd

import (
	"encoding/json"
	"fmt"

	"github.com/spf13/cobra"

	"mm-cli/internal/apps"
	"mm-cli/internal/manifest"
)

func runManifest(cmd *cobra.Command, args []string) error {
	if len(args) > 0 && args[0] == "help" {
		return cmd.Help()
	}
	refresh, _ := cmd.Flags().GetBool("refresh")
	wantJSON, _ := cmd.Root().PersistentFlags().GetBool("json")
	ctx := cmd.Context()

	if len(args) == 0 {
		// All apps, summary.
		type row struct {
			Slug    string `json:"slug"`
			Version string `json:"version,omitempty"`
			Actions int    `json:"actions"`
			Err     string `json:"err,omitempty"`
		}
		rows := make([]row, 0, len(apps.Registry))
		for slug := range apps.Registry {
			r := row{Slug: slug}
			m, err := manifest.Load(ctx, slug, refresh)
			if err != nil {
				r.Err = err.Error()
			} else {
				r.Version = m.Version
				r.Actions = m.ActionCount()
			}
			rows = append(rows, r)
		}
		if wantJSON {
			out, _ := json.MarshalIndent(rows, "", "  ")
			fmt.Println(string(out))
			return nil
		}
		fmt.Println("# Manifests")
		for _, r := range rows {
			if r.Err != "" {
				fmt.Printf("  %s  (fetch failed: %s)\n", padRight(r.Slug, 12), r.Err)
				continue
			}
			fmt.Printf("  %s  v%s  %d actions\n", padRight(r.Slug, 12), r.Version, r.Actions)
		}
		return nil
	}

	m, err := manifest.Load(ctx, args[0], refresh)
	if err != nil {
		return err
	}
	if wantJSON {
		out, _ := json.MarshalIndent(m, "", "  ")
		fmt.Println(string(out))
		return nil
	}
	fmt.Printf("# %s v%s\n\n", m.AppSlug, m.Version)
	for feature, actions := range m.Features {
		fmt.Printf("## %s (%d)\n", feature, len(actions))
		for action, def := range actions {
			auth := def.Auth
			if auth == "" {
				auth = "?"
			}
			desc := ""
			if def.Description != "" {
				desc = "  — " + def.Description
			}
			fmt.Printf("  %s.%s  [auth: %s]%s\n", feature, action, auth, desc)
		}
		fmt.Println()
	}
	return nil
}

File: src/commands/manifest.ts
// ⚠️ LEGACY TYPESCRIPT PORT — NOT the live `mm` binary. The live CLI is Go;
// this command lives in internal/cmd/. Editing this file changes nothing in
// `mm` (it only builds the separate, unused `mm-ts`). Fix the .go file instead.

/**
 * `mm manifest <app>` — print an app's manifest, summary or full.
 *
 *   mm manifest kb              # one-line per feature, action count
 *   mm manifest kb --json       # raw manifest JSON
 *   mm manifest kb --refresh    # bypass 24h cache
 *   mm manifest                 # list all apps + their cached state
 */

import { loadManifest, actionCount } from '../manifest';
import { listApps, resolveApp } from '../apps';

export async function manifestDispatch(
	args: string[],
	flags: { json?: boolean; refresh?: boolean }
) {
	const slug = args[0];

	if (!slug) {
		// Show all apps with their cached state.
		console.log('Known apps:');
		for (const app of listApps()) {
			console.log(`  ${app.slug.padEnd(12)} ${app.url}  — ${app.description ?? ''}`);
		}
		console.log('');
		console.log('Use: mm manifest <slug>          to fetch + show one app');
		console.log('     mm manifest <slug> --json   for raw JSON');
		console.log('     mm manifest <slug> --refresh to bypass 24h cache');
		return;
	}

	resolveApp(slug); // throws if unknown
	const manifest = await loadManifest(slug, { refresh: flags.refresh });

	if (flags.json) {
		console.log(JSON.stringify(manifest, null, 2));
		return;
	}

	const total = actionCount(manifest);
	console.log(`${manifest.appSlug} (${manifest.version}) — ${total} actions across ${Object.keys(manifest.features).length} features`);
	console.log('');

	const features = Object.entries(manifest.features).sort(([a], [b]) => a.localeCompare(b));
	for (const [feature, actions] of features) {
		const actionEntries = Object.entries(actions).sort(([a], [b]) => a.localeCompare(b));
		console.log(`  ${feature}`);
		for (const [action, def] of actionEntries) {
			const desc = def.description ? `  — ${def.description}` : '';
			console.log(`    ${feature}.${action.padEnd(30)} [${def.auth}]${desc}`);
		}
	}
	console.log('');
	console.log(`Use: mm v2 ${slug} <feature.action> [json-payload]   to invoke`);
}